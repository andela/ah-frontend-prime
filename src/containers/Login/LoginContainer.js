import React, { Component } from "react";
import { connect } from "react-redux";

import LoginModal from "../../components/login/LoginModal";
import { userLoginRequest } from "../../actions/LoginAction";

export class LoginContainer extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: "",
      password: "",
      isLoading: false,
      errors: {
        email: "",
        password: ""
      }
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.loginSuccess === true) {
      const { history } = this.props;

      history.push("/");
    }
  }

  handleChange(event) {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  handleSubmit(event) {
    event.preventDefault();

    this.setState({
      errors: {
        email: "The email is required"
      }
    });
    const data = {
      user: {
        email: this.state.email,
        password: this.state.password
      }
    };
    this.props.userLoginRequest(data);
  }

  render() {
    return (
      <div>
        <LoginModal
          handleChange={this.handleChange}
          email={this.state.email}
          isProcessing={this.props.isProcessing}
          password={this.state.password}
          errors={this.state.errors}
          handleSubmit={this.handleSubmit}
        />
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  isProcessing: state.auth_login.isProcessing,
  loginSuccess: state.auth_login.loginSuccess
});

export default connect(
  mapStateToProps,
  { userLoginRequest }
)(LoginContainer);
