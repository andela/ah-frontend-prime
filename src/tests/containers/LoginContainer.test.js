import React from "react";
import configureStore from "redux-mock-store";
import { shallow, mount } from "enzyme";
import thunk from "redux-thunk";
import moxios from "moxios";
import store from "../../store";
import LoginContainer, {
  LoginContainer as DumpLoginContainer,
  mapStateToProps
} from "../../containers/Login/LoginContainer";

describe("login container", () => {
  const initialState = {
    login: { isSuccesfull: false, token: "", errors: null }
  };
  const props = {
    history: { push: jest.fn() }
  };

  const nextProps = {
    loginSuccess: true
  };
  const mockFn = jest.fn();
  const mockStore = configureStore([thunk]);
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    moxios.install();
    wrapper = shallow(
      <DumpLoginContainer open={false} close={mockFn} {...props} />
    );
  });

  afterEach(function() {
    moxios.uninstall();
  });
  it("should render without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should handle the onChange event", () => {
    const event = {
      target: {
        name: "password",
        value: "pass123"
      }
    };
    wrapper.instance().handleChange(event);
    expect(wrapper.instance().state.password).toBe("pass123");
  });
  

  it("should handle the onSubmit event", () => {
    const props = {
      userLoginRequest: jest.fn()
    };
    let wrapper = shallow(<DumpLoginContainer {...props} />);
    const instance = wrapper.instance();
    const event = {
      target: {
        name: "",
        value: "Hello"
      }
    };
    event.preventDefault = jest.fn();
    instance.handleSubmit(event);
    expect(instance.state.errors.email).toEqual("The email is required");
  });

  it("should mapStateToProps", () => {
    const state = {
      auth_login: { isProcessing: false }
    };
    expect(mapStateToProps(state)).toEqual({ isProcessing: false });
  });
  it("should not redirect on failed login", () => {
    wrapper.setProps({ loginSuccess: false });

    expect(props.history.push).toBeCalledTimes(0);
  });

  it("should redirect on login success", () => {
    wrapper.setProps({ ...nextProps });

    expect(props.history.push).toBeCalledWith("/");
  });
});
