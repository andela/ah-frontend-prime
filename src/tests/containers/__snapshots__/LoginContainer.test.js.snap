// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login container should render without crashing 1`] = `ShallowWrapper {}`;
